pkg/resources/fluentd/appconfigmap.go

```diff
@@ -77,6 +77,16 @@ func (r *Reconciler) hasConfigCheckPod(ctx context.Context, hashKey string) (boo
 	return true, nil
 }
 
+func (r *Reconciler) CheckForObjectExistence(ctx context.Context, object client.Object) (*ConfigCheckResult, error) {
+	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(object), object); err != nil {
```

Comment:
In case the error is `apierrors.IsNotFound(err)` we could also return without an error, since we will retry the reconcile once the secret update is received automatically. What do you think?

Reply:
I agree, thanks for it!